core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -741,4 +798,25 @@ public Executor getPersistenceExecutor() {
    public StreamingMarshaller getMarshaller() {
       return m;
    }
+
+   private void removeCacheLoader(String storeType, Collection<CacheLoader> collection) {
+      collection.removeIf(cacheLoader -> undelegate(cacheLoader).getClass().getName().equals(storeType));
+   }
+
+   private void removeCacheWriter(String storeType, Collection<? extends CacheWriter> collection) {
+      collection.removeIf(cacheWriter -> undelegate(cacheWriter).getClass().getName().equals(storeType));
+   }
+
+   private void performOnAllTxStores(Predicate<CacheWriter> filter, Consumer<TransactionalCacheWriter> action) {
+      storesMutex.readLock().lock();
+      try {
+         txWriters.stream().map(TransactionalCacheWriter.class::cast).filter(filter).forEach(action);
```

Comment:
Also, the filter predicate seems a bit excessive, since you'll always filter on the access mode. And I'm not sure escape analysis can eliminate the extra allocations...